docs/modules/ROOT/pages/gsn.adoc

```diff
@@ -0,0 +1,77 @@
+
+**Enter meta-transactions**. This is a fancy name for a simple idea: a third-party can send another user's transactions and pay themselves for the gas cost. That's it! There's some tricky technical details, but those can be safely ignored when interacting with the GSN. This way, instead of your users calling into your contract (called the _recipient_) directly, someone else (we'll call them a _relay_) will send their transaction and pay for the cost.
+
+But why would they do such a thing?
+
+== Incentives
+
+Relays are not running a charity: they're running a bussiness. The reason why they'll gladly pay for your users' gas costs is because they will in turn charge your contract, the recipient. That way relays get their money back, plus a bit extra as a _fee_ for their services.
+
+This may sound strange at first, but paying for user onboarding is a very common bussiness practice. Lots of money are spent on advertising, free trials, new user discounts, etc., all with the goal of user acquisition. And compared to those, the cost of a couple Ethereum transactions is actually very small.
+
+Additionally, you can leverage the GSN in scenarios where your users pay you off-chain in advance (e.g. via credit card), with each GSN-call deducting from their balance on your system. The possibilities are endless!
+
+=== Should I trust these relays?
+
+You don't need to! The GSN is set up in such a way where it's in the relays' best interest to serve your requests, and there are measures in place to penalize them if they misbeheave. All of this happens automatically, so you can safely start using their services worry-free.
+
+== One contract to coordinate them all
+
+There are many meta-transaction implementations out there, but the GSN has a unique detail that makes it special. Sitting in the middle of the system, a smart contract is charged with keeping track of relays, handling relayed transactions, charging their recipients, and generally ensuring all parties stay honest. This contract is called RelayHub, and there is a _single_ instance of it in the whole network (you don't need to deploy your own!). Think of it as a piece of public infrastructure, for all Ethereum users to benefit from.
+
+One of RelayHub's tasks is relay-coordination: all relays will advertise their services on this contract, and your users will query it to find the relay that best suits their purposes. This is out of scope for this guide however, and is not something you need to worry about when writing a recipient contract. If you want to learn more about sending transactions via relays, head to our [GSNProvider guide]().
+
+== Receiving a relayed call
+
+The contract you will be inheriting from is GSNRecipient. It includes the IRelayRecipient interface, plus a number of useful utilities that handle the low-level details of the GSN, so you can focus on your dApp.
```

Comment:
This section feels disconnected from the previous ones, and its beginning is a bit harsh. Why do I want to inherit from what? We should add a bit more context: tell the reader that we will now explain how to add GSN support in their app.

Reply:
I first wrote this bit and didn't review the transition when I added the intro, it did end up quite jarring. I'll add a couple sentences for context.